core/src/main/java/org/apache/spark/shuffle/api/ShufflePartitionWriter.java

```diff
@@ -85,14 +85,4 @@
   default Optional<WritableByteChannelWrapper> openChannelWrapper() throws IOException {
     return Optional.empty();
   }
-
-  /**
-   * Returns the number of bytes written either by this writer's output stream opened by
-   * {@link #openStream()} or the byte channel opened by {@link #openChannelWrapper()}.
-   * <p>
-   * This can be different from the number of bytes given by the caller. For example, the
-   * stream might compress or encrypt the bytes before persisting the data to the backing
-   * data store.
-   */
-  long getNumBytesWritten();
```

Comment:
I think we _might- still need this - appears to be used by the sort shuffle writer per https://github.com/apache/spark/pull/25342/files#diff-fe378a929dd1f5c5ac8bff90dab743b1R87... hmm.

Reply:
Could you update the metrics after the commit (by adding up all the partition lengths)?

Otherwise it doesn't seem horrible to keep this in the API.

Reply:
Let's just keep the API for now.